### DISCUSSION TITLE
Custom hash for task dependecies

### DISCUSSION THREAD

**Discussion by stormojm** [vercel/turborepo#1720] [Ideas]
In the case that you have a rather large typescript project, you may change one of your common dependencies which will require a rebuild of most/all of your other typescript libraries.  This includes if you only made a non-breaking API change.  What you really want to do in this case is only rebuild if your dependency changes its API (i.e. their .d.ts files).

You can't just change the outputs, because you still want to cache a packages transpiled source.

NX supports a similar concept in their custom executors: [ Link to NX docs](https://nx.dev/using-nx/caching)

> This behavior is customizable. For instance, lint checks may only depend on the source code of the project and global configs. Builds can depend on the dts files of the compiled libs instead of their source.

**Comment by anthonyshew** [comment]
This doesn't sound like something we will want to support. The runtime behavior of a package can change despite the API not changing. In that case, it would be appropriate for the dependent packages and applications to rebuild.
